### ISSUE TITLE
[Bug]  failed to load ai detrction model

### ISSUE THREAD

**Issue by mychaow** [kaya-go/kaya#88]
## Description

cannot scan a image

<img width="2742" height="1754" alt="Image" src="https://github.com/user-attachments/assets/0f484143-67bc-4d63-988a-3aa47b9c0e6b" />

## Steps to Reproduce

2. Click on '...'
3. See error

## Expected Behavior

What you expected to happen.

## Actual Behavior

What actually happened.

## Environment

- **Platform**: Desktop
- **OS**: macOS 14.4.1
- **Browser** (web only): (e.g., Chrome 120, Firefox 121)
- **Kaya Version**: 0.3.8

## Additional Context

- Does this happen with a specific SGF file? If so, please attach it.
- Does this happen on a specific board size (9x9, 13x13, 19x19)?
- Any console errors? (Press F12 to open developer tools)

with a image? 

<img width="509" height="540" alt="Image" src="https://github.com/user-attachments/assets/c9c2aebb-5a84-4d6d-adfb-8dda1b2f52e2" />

**Comment by hadim** [comment]
Can you provide the logs from the console when this happens using  Cmd+Option+I (macOS) ou Ctrl+Shift+I (Linux/Windows) ?

The detection model is downloaded from https://huggingface.co/kaya-go/moku-v3/tree/main so maybe check whether it works as well.

**Comment by hadim** [comment]
I will soon a release a new version that will bundle the detection model directly into the desktop app. That should solve your issue.

**Comment by hadim** [comment]
Please try the new version, 0.3.9 at https://github.com/kaya-go/kaya/releases

The model is now bundled in the desktop app, so it should resolve your issue. Please reopen with console logs if you still having the issue.

**Comment by mychaow** [comment]
Hi, @hadim Thanks for your reply, I try the new version, but get error like this.

<img width="3456" height="2160" alt="Image" src="https://github.com/user-attachments/assets/cc645388-f1be-405d-91f8-6a5b5a9f1297" />

`Failed to load AI detection model: Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 void onnxruntime::Graph::InitializeStateFromModelFileGraphProto() This is an invalid model. Graph output (logits) does not exist in the graph. `

**Comment by hadim** [comment]
Hi @mychaow, thanks for testing v0.3.9!

The error Graph output (logits) does not exist in the graph means the ONNX model file that was loaded doesn't match the expected format. On my end (same version, macOS), the bundled model loads correctly:

```
[Moku] Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.6s
[Moku] Ready in 1.7s
```

This could happen if the app is loading a stale or corrupted model file. Could you try the following:

- Open DevTools (Cmd+Option+I on macOS or Ctrl+Shift+I on Linux/Windows)
- Go to Application → Cache Storage → delete any kaya-moku-models cache entries
- Also check Application → IndexedDB → delete any stored model data
- Restart the app and try again

If the issue persists, please share the full console logs from DevTools when you trigger the board scan — I need to see whether it says Model loaded (bundled) or something else, to pinpoint where the bad model is coming from.

**Comment by mychaow** [comment]
<img width="3456" height="1988" alt="Image" src="https://github.com/user-attachments/assets/928a3578-ab81-4b9a-851f-2619fa0ea67d" /> 

Hi, @hadim thank you for your patient reply, but I get the new error like this. It seems that broad size is not right, but and I can not change broard size to rescan it.   I have tried the kayaweb.app，it can work well.

**Comment by mychaow** [comment]
[Error] [BoardRecognition] Moku init failed: – "Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/framework/node_index_info.cc:88 aut…"
"Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/framework/node_index_info.cc:88 auto onnxruntime::NodeIndexInfo::Init(const onnxruntime::ValidNodes<const std::vector<const onnxruntime::Node *>> &, NodeIndex, const OrtValueNameIdxMap &)::(lambda)::operator()(const onnxruntime::NodeArg &, bool) const status.IsOK() was false. Could not find OrtValue with name '/model/model/Shape_78_output_0'
"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:6:22047)
[Error] [BoardRecognition] Moku init failed: – "Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 voi…"
"Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 void onnxruntime::Graph::InitializeStateFromModelFileGraphProto() This is an invalid model. Graph output (logits) does not exist in the graph.
"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:6:22047)
[Error] [GLOBAL ERROR] – "{↵  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",↵  \"source\": \"tauri://localhost\",↵  \"lineno\": 0,↵  \"colno\"…"
"{
  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",
  \"source\": \"tauri://localhost\",
  \"lineno\": 0,
  \"colno\": 0,
  \"error\": \"null\"
}"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:16:173)
[Error] ResizeObserver loop completed with undelivered notifications. (localhost, line 0)
[Error] [GLOBAL ERROR] – "{↵  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",↵  \"source\": \"tauri://localhost\",↵  \"lineno\": 0,↵  \"colno\"…"
"{
  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",
  \"source\": \"tauri://localhost\",
  \"lineno\": 0,
  \"colno\": 0,
  \"error\": \"null\"
}"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:16:173)
[Error] ResizeObserver loop completed with undelivered notifications. (localhost, line 0)
[Error] [GLOBAL ERROR] – "{↵  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",↵  \"source\": \"tauri://localhost\",↵  \"lineno\": 0,↵  \"colno\"…"
"{
  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",
  \"source\": \"tauri://localhost\",
  \"lineno\": 0,
  \"colno\": 0,
  \"error\": \"null\"
}"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:16:173)
[Error] ResizeObserver loop completed with undelivered notifications. (localhost, line 0)
[Error] [GLOBAL ERROR] – "{↵  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",↵  \"source\": \"tauri://localhost\",↵  \"lineno\": 0,↵  \"colno\"…"
"{
  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",
  \"source\": \"tauri://localhost\",
  \"lineno\": 0,
  \"colno\": 0,
  \"error\": \"null\"
}"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:16:173)
[Error] ResizeObserver loop completed with undelivered notifications. (localhost, line 0)
[Error] [BoardRecognition] Moku init failed: – "Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 voi…"
"Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 void onnxruntime::Graph::InitializeStateFromModelFileGraphProto() This is an invalid model. Graph output (logits) does not exist in the graph.
"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:6:22047)
[Error] [GLOBAL ERROR] – "{↵  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",↵  \"source\": \"tauri://localhost\",↵  \"lineno\": 0,↵  \"colno\"…"
"{
  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",
  \"source\": \"tauri://localhost\",
  \"lineno\": 0,
  \"colno\": 0,
  \"error\": \"null\"
}"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:16:173)
[Error] ResizeObserver loop completed with undelivered notifications. (localhost, line 0)
[Error] [GLOBAL ERROR] – "{↵  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",↵  \"source\": \"tauri://localhost\",↵  \"lineno\": 0,↵  \"colno\"…"
"{
  \"message\": \"ResizeObserver loop completed with undelivered notifications.\",
  \"source\": \"tauri://localhost\",
  \"lineno\": 0,
  \"colno\": 0,
  \"error\": \"null\"
}"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:16:173)
[Error] ResizeObserver loop completed with undelivered notifications. (localhost, line 0)
[Log] [Moku] – "Initializing…" (141.f230b58f.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.8s" (141.f230b58f.js, line 2)
[Log] [Moku] – "Initializing…" (141.f230b58f.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.8s" (141.f230b58f.js, line 2)
[Log] [Moku] – "Initializing…" (141.f230b58f.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.8s" (141.f230b58f.js, line 2)
[Log] [Moku] – "Initializing…" (141.f230b58f.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.8s" (141.f230b58f.js, line 2)
[Log] [Moku] – "Initializing…" (141.f230b58f.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.8s" (141.f230b58f.js, line 2)
[Error] [BoardRecognition] Moku init failed:
"Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 void onnxruntime::Graph::InitializeStateFromModelFileGraphProto() This is an invalid model. Graph output (logits) does not exist in the graph.
"
	(anonymous function) (index.5b973597.js:16:682)
	(anonymous function) (index.5b973597.js:6:22047)
[Log] [Moku] – "Initializing…" (141.f230b58f.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.9s" (141.f230b58f.js, line 2, x2)

**Comment by hadim** [comment]
Can you try the latest version, please? https://github.com/kaya-go/kaya/releases/tag/v0.3.10

And report the logs if failing again please.

**Comment by mychaow** [comment]
```
[Log] Version info: (index.d484e455.js, line 75)
Object

buildDate: "2026-04-05T08:11:31.998Z"

gitHash: "d3e9f46"

version: "0.3.10"

Object Prototype
[Error] [BoardRecognition] Moku init failed:
"Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 void onnxruntime::Graph::InitializeStateFromModelFileGraphProto() This is an invalid model. Graph output (logits) does not exist in the graph.
"
	(anonymous function) (index.d484e455.js:16:682)
	(anonymous function) (index.d484e455.js:6:22047)
[Log] [Moku] – "Initializing…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.9s" (141.6de57ac4.js, line 2)
[Log] [Moku] – "graphOptimizationLevel 'all' not supported, trying next level…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "graphOptimizationLevel 'basic' not supported, trying next level…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "Initializing…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.9s" (141.6de57ac4.js, line 2)
[Log] [Moku] – "graphOptimizationLevel 'all' not supported, trying next level…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "graphOptimizationLevel 'basic' not supported, trying next level…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "Initializing…" (141.6de57ac4.js, line 2)
[Error] error sending request for url (https://github.com/kaya-go/kaya/releases/latest/download/latest.json)
	(anonymous function) (index.d484e455.js:16:682)
	(anonymous function) (index.d484e455.js:7:191917)
[Log] [Moku] – "Model loaded (bundled) from /models/moku-v3.onnx: 77.1 MB in 0.8s" (141.6de57ac4.js, line 2)
[Log] [Moku] – "graphOptimizationLevel 'all' not supported, trying next level…" (141.6de57ac4.js, line 2)
[Log] [Moku] – "graphOptimizationLevel 'basic' not supported, trying next level…" (141.6de57ac4.js, line 2)
[Error] [BoardRecognition] Moku init failed:
"Can't create a session. ERROR_CODE: 1, ERROR_MESSAGE: /mnt/vss/_work/1/s/onnxruntime/core/graph/graph.cc:1494 void onnxruntime::Graph::InitializeStateFromModelFileGraphProto() This is an invalid model. Graph output (logits) does not exist in the graph.
"
	(anonymous function) (index.d484e455.js:16:682)
	(anonymous function) (index.d484e455.js:6:22047)
[Error] error sending request for url (https://github.com/kaya-go/kaya/releases/latest/download/latest.json)
	(anonymous function) (index.d484e455.js:16:682)
	(anonymous function) (index.d484e455.js:7:191917)
```

**Comment by hadim** [comment]
Once done, can you try the build at https://github.com/kaya-go/kaya/actions/runs/24002508950. You can download the dmg at the bottom<

**Comment by hadim** [comment]
0.3.11 also contains the fix https://github.com/kaya-go/kaya/releases/tag/v0.3.11

**Comment by hadim** [comment]
Closing here but please re open if needed.
